### PR TITLE
redfish: Fetch the inventory member correctly

### PR DESCRIPTION
**Pull request by lcp**
In the Members array of FirmwareInventory or SoftwareInventory, each
element of the array only contains the URI to the member. For example:

  "Members":
  [
    {
      "@odata.id": "/redfish/v1/UpdateService/FirmwareInventory/1/"
    },
    {
      "@odata.id": "/redfish/v1/UpdateService/FirmwareInventory/2/"
    },
    {
      "@odata.id": "/redfish/v1/UpdateService/FirmwareInventory/3/"
    }
  ]

We have to get the real member object through the given URI.

### CODE REVIEW

File: plugins/redfish/fu-redfish-client.c [hunk 1]
```diff
@@ -100,11 +100,69 @@ fu_redfish_client_coldplug_inventory (FuRedfishClient *self,
...
+		g_autoptr(GBytes) blob = NULL;
+
+		member_id = json_array_get_object_element (members, i);
+		member_uri = json_object_get_string_member (member_id, "@odata.id");
+		if (member_uri == NULL) {
+			g_set_error_literal (error,
+					     FWUPD_ERROR,
+					     FWUPD_ERROR_NOT_FOUND,
+					     "no @odata.id string");
+			g_object_unref (parser);
```
Comment (hughsie) [inline plugins/redfish/fu-redfish-client.c]: you never need to unref the parser, the `g_autoptr(JsonParser)` does this for you when it goes out of scope.

File: plugins/redfish/fu-redfish-client.c [hunk 2]
```diff
@@ -100,11 +100,69 @@ fu_redfish_client_coldplug_inventory (FuRedfishClient *self,
 				      GError **error)
 {
 	JsonArray *members;
+	JsonObject *member_id;
+	const gchar *member_uri;
```
Comment (hughsie) [inline plugins/redfish/fu-redfish-client.c]: You can move these inside the loop, thanks.

File: plugins/redfish/fu-redfish-client.c (hunk 1 above)
Comment (lcp) [reply plugins/redfish/fu-redfish-client.c]: Oh, great. Nice to know that :D

File: plugins/redfish/fu-redfish-client.c (hunk 2 above)
Comment (lcp) [reply plugins/redfish/fu-redfish-client.c]: Ok, I'll push a refreshed patch.

**Comment by lcp** [comment]
Is the updated patch good for you?

**Comment by hughsie** [comment]
Sorry, am on holiday. Merged just now, thanks.

**Pull request merged**